dev.Dockerfile

```diff
@@ -0,0 +1,61 @@
+# Note: in future we should use the timescaledb-ha image. Unfortunately it
+# doesn't have arm64 builds, so we're doing things from scratch.
+FROM ubuntu:22.04
+
+SHELL ["/bin/bash", "-eE", "-o", "pipefail", "-c"]
+
+RUN apt update && apt install -y sudo wget curl gnupg2 lsb-release
+
+# Setup a non-root user that we'll use
+RUN adduser --disabled-password --gecos "" ubuntu && \
+ usermod -aG sudo ubuntu && \
+ echo "ubuntu ALL=(ALL:ALL) NOPASSWD: ALL" > /etc/sudoers.d/ubuntu
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Install timescaledb
+RUN echo "deb http://apt.postgresql.org/pub/repos/apt/ $(lsb_release -c -s)-pgdg main" > /etc/apt/sources.list.d/pgdg.list
+RUN wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
+RUN echo "deb [signed-by=/usr/share/keyrings/timescale.keyring] https://packagecloud.io/timescale/timescaledb/ubuntu/ $(lsb_release -c -s) main" > /etc/apt/sources.list.d/timescaledb.list
+RUN wget --quiet -O - https://packagecloud.io/timescale/timescaledb/gpgkey | gpg --dearmor -o /usr/share/keyrings/timescale.keyring
+RUN apt-get update && apt-get install -y timescaledb-2-postgresql-{12,13,14}
+RUN apt-get install -y build-essential clang libssl-dev pkg-config libreadline-dev zlib1g-dev postgresql-server-dev-{12,13,14}
+
+# These directories need to be writeable for pgx to install the extension into
+RUN chmod a+w /usr/share/postgresql/*/extension /usr/lib/postgresql/*/lib
+
+USER ubuntu
+
+# Install rust
+ENV RUST_VERSION=1.62.0
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --no-modify-path --profile minimal --component rustfmt --default-toolchain ${RUST_VERSION}
+ENV PATH=/home/ubuntu/.cargo/bin:$PATH
+
+RUN cargo install cargo-pgx --git https://github.com/timescale/pgx --branch promscale-staging --rev ee52db6b
+
+RUN cargo pgx init --pg14 /usr/lib/postgresql/14/bin/pg_config --pg13 /usr/lib/postgresql/13/bin/pg_config --pg12 /usr/lib/postgresql/12/bin/pg_config
```

Comment:
We install various PG versions but it seems we only use PG14. If there is not easy way to use different PG version maybe better only have PG14.

Reply:
It is easy to use them, but right now it requires some manual intervention. I'm a little on the fence as to what to do here. I'm not sure how much of a use-case it is to run different postgres versions, but I'm sure that somebody will need to test something against pg12 because it behaves differently for some reason. I think what I will do is make it so that you can set `PG_VERSION` in the container's env, and it will run that version. WDYT?

(Initially I was thinking that it would be cool to run all postgres versions and expose them all on different ports, but then when recompiling, we would effectively have to recompile 3 times, which seems like it's not worth the effort).